compiler/rustc_metadata/src/locator.rs

```diff
@@ -1208,10 +1219,25 @@ impl CrateError {
                 "plugin `{}` only found in rlib format, but must be available in dylib format",
                 crate_name,
             ),
-        };
+        }
+    }
 
-        err.emit();
+    crate fn report(self, sess: &Session, span: Span, missing_core: bool) {
+        self.build_diag(sess, span, missing_core).emit();
+    }
+
+    crate fn report_fatal(self, sess: &Session, span: Span, missing_core: bool) -> ! {
+        self.report(sess, span, missing_core);
         sess.abort_if_errors();
         unreachable!();
```

Comment:
`abort_if_erros` + `unreachable` can be replaced with just `FatalError.raise()`, and I'd just remove this function after that.